mne/io/eyetrack/eyetrack.py

```diff
@@ -0,0 +1,243 @@
+# Author: Dominik Welke <dominik.welke@web.de>
+#
+# License: BSD-3-Clause
+
+import numpy as np
+
+from ..base import BaseRaw
+from ..meas_info import create_info
+from ..constants import FIFF
+
+from ...annotations import Annotations
+from ...utils import logger, verbose, fill_doc, warn
+
+
+@fill_doc
+def read_raw_eyelink(fname, preload=False, verbose=None,
+                     annotate_missing=False, interpolate_missing=False,
+                     read_eye_events=True):
+    """Reader for an XXX file.
+
+    Parameters
+    ----------
+    fname : str
+        Path to the XXX data file (.XXX).
+    %(preload)s
+    %(verbose)s
+
+    Returns
+    -------
+    raw : instance of RawEyetrack
+        A Raw object containing eyetracker data.
+
+    See Also
+    --------
+    mne.io.Raw : Documentation of attribute and methods.
+    """
+    return RawEyelink(fname, preload=preload, verbose=verbose,
+                      annotate_missing=annotate_missing,
+                      interpolate_missing=interpolate_missing,
+                      read_eye_events=read_eye_events)
+
+
+@fill_doc
+class RawEyelink(BaseRaw):
+    """Raw object from an XXX file.
+
+    Parameters
+    ----------
+    fname : str
+        Path to the data file (.XXX).
+    %(preload)s
+    %(verbose)s
+
+    See Also
+    --------
+    mne.io.Raw : Documentation of attribute and methods.
+    """
+
+    @verbose
+    def __init__(self, fname, preload=False, verbose=None,
+                 annotate_missing=True, interpolate_missing=True,
+                 read_eye_events=True):
+        from ...preprocessing import annotate_nan
+        from ...preprocessing.interpolate import interpolate_nan
+
+        logger.info('Loading {}'.format(fname))
+
+        # load data
+        ftype = fname.split('.')[-1]
+        if ftype == 'asc':
+            sfreq = 500
+            eye = 'BINO'
+            pos = True
+            pupil = True
+
+            info, data, first_sample, annot, meas_date = \
+                self._parse_eyelink_asc(
+                    fname,
+                    sfreq=sfreq,
+                    eye=eye,
+                    pos=pos,
+                    pupil=pupil,
+                    read_eye_events=read_eye_events)
+        elif ftype == 'edf':
+            raise NotImplementedError('Eyelink .edf files not supported, yet')
```

Comment:
I removed this line - IMO I don't think support for reading eyelink data files (.edf) directly is feasible. This would require the user to have the Eyelink SDK properly installed on their machine, and MNE-Python codebase would need wrapper functions to execute the eyelink SDK binary files (and these functions would need to infer the user machine's OS, and act accordingly). My understanding is that this is something that can be very difficult to maintain, it is probably more simple for users to convert their files to .asc format with the tools eyelink provides. but feel free to ask the core devs, if you disagree.

Reply:
At some point pyeparse supported EDF I think with this code:

https://github.com/pyeparse/pyeparse/tree/master/pyeparse/edf

If it still works we could try using it. But it can be in a follow-up PR since it might indeed be a pain, and adding ASCII support is already a good first step.

Reply:
@larsoner Oh okay maybe I was mistaken then, sounds good!

Reply:
My concern wasn't having to use another software (like using eyelinks edf2asc converter) so much as having to maintain a wrapper_function in `mne` that accesses eyelink's API. But if the pyparse code is already there, maybe it's not so bad as I thought!

Weird that it only works with monocualr recordings. I have some monocular recordings I can test out with it.